src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -63,6 +65,18 @@ public Pose2d getPose() {
         return inputs.pose;
     }
 
+    public ChassisSpeeds getRobotSpeeds() {
+        return inputs.chassisSpeeds;
+    }
+
+    public RobotConfig getRobotConfig() {
+        return constants.robotConfig;
+    }
+
+    public double getMaxDriveVelocityMPS() {
+        return constants.maxDriveVelocityMPS;
+    }
```

Comment:
These should probably not be publicly exposed. If things need to follow a path, they should use a command factory that can be added in a separate PR.